src/spaceTruckerApplication.js

```diff
@@ -0,0 +1,123 @@
+
+import AppStates from "./appstates"
+import logger from "./logger"
+import MainMenuScene from "./mainMenuScene";
+
+
+class SpaceTruckerApplication {
+    *appStateMachine() {
```

Comment:
did you mean to create a generator?

Reply:
oh, yes you did :-)

Reply:
simple iterator-based finite state machine implementation. I made an experimental branch that uses a fsm implementation I found on npm, but I think it might be overkill for this situation